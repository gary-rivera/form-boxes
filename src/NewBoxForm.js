import React, { useState } from 'react';

/**
 * Props:
 *  - create function 
 * State:
 *  - formData
 */
const initialState = {
  width: "",
  height: "",
  color: ""
};

function NewBoxForm({ create }) {
  const [formData, setFormData] = useState(initialState);

  function handleChange(evt) {
    const { name, value } = evt.target;
    
    setFormData( fData => ({...fData, [name]: value }));
  }
  
  function handleSubmit(evt) {
    evt.preventDefault();

    create(formData);
    setFormData(initialState);
  }

  return (
  <div>
    <form onSubmit={handleSubmit}> 
      <label htmlFor="height">Height</label> 
      <input id="height" 
        onChange={handleChange} 
        name="height" 
        value={formData.height}/>

      <label htmlFor="width">Width</label> 
      <input id="width" 
        onChange={handleChange} 
        name="width" 
        value={formData.width}/>

      <label htmlFor="color">Color</label> 
      <input id="color" 
        onChange={handleChange} 
        name="color"
        value={formData.color}/>
      <button>Add box</button>
    </form>
  </div>
  );
}

export default NewBoxForm;