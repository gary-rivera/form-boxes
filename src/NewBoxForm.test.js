import {fireEvent, render} from '@testing-library/react'
import NewBoxForm from './NewBoxForm'

it("renders without crashing", function() {
    render(<NewBoxForm/>)
})

it("matches initial snapshot", function() {
    const { container } = render(<NewBoxForm/>)
    expect(container).toMatchSnapshot();
})

it("submitting works", function() {
    const mockBox = jest.fn();
    const { container } = render(<NewBoxForm create={mockBox}/>)
    const height = container.querySelector('input[name="height"]');
    const width = container.querySelector('input[name="width"]');
    const color = container.querySelector('input[name="color"]');

    fireEvent.change(height, {target: {value:'50'}});
    fireEvent.change(width, {target: {value:'50'}});
    fireEvent.change(color, {target: {value:'blue'}});

    expect(height).toHaveValue('50');
    expect(width).toHaveValue('50');
    expect(color).toHaveValue('blue');

    const button = container.querySelector('button');
    fireEvent.click(button);
    expect(height).toHaveValue('');
    expect(width).toHaveValue('');
    expect(color).toHaveValue('');

})

