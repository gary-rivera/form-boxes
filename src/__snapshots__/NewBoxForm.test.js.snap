// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches initial snapshot 1`] = `
<div>
  <div>
    <form>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        value=""
      />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        value=""
      />
      <button>
        Add box
      </button>
    </form>
  </div>
</div>
`;
